javascript/javascript-docs/docs/high-level-design-fe.md

```diff
@@ -63,11 +63,26 @@ TBD
 
 ## Where should the business logic reside?
 
-TBD
+Most of our application uses Redux, therefor this section will focus on where to locate logic using redux.
+In redux there are actions and reducers. A reducer is a pure function which returns the same value for a specific input and has no side effects.
+We also use Selectors ([reselect](https://github.com/reduxjs/reselect)) to fetch data from the store.
+In a react redux application, the data should be kept on the store.
+We prefer putting as much logic in the reducer rather than the selector/action/event handler who started the operation.
```

Comment:
shouldn't the business logic be on the middleware?

Reply:
Well, I guess we can talk about it but I believe that redux-logic messes up the clear separation of concerns. A middleware is only a tool for us to achieve something, IMO it shouldn't contain real logic, but that's my opinion, lets discuss it :)

Reply:
Basically, I believe in what redux claim in their docs:
https://redux.js.org/style-guide/style-guide#put-as-much-logic-as-possible-in-reducers

Reply:
yeah I agree, if its logic related tot he store than it should be in the reducer, I think I got confused with different kind of logic like async logic https://redux.js.org/style-guide/style-guide#use-thunks-for-async-logic

Reply:
Agreed, async operations should happen within a middleware, definitely.